account_invoice_hour_overview/account_invoice.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# © 2017 Sunflower IT (http://sunflowerweb.nl)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from itertools import groupby
+import time
+from datetime import datetime as dt
+
+from openerp import fields, models, api, _
+import openerp.addons.decimal_precision as dp
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def _get_report_analytic_lines(self):
+        for invoice in self:
+            lines = self.env['hr.analytic.timesheet'].search([
+                ('invoice_id', '=', invoice.id)
+            ])
+            invoice.report_analytic_lines = lines
+
+    report_analytic_lines = fields.One2many('hr.analytic.timesheet',
+        string='Analytic lines',
+        help="The analytic lines coupled to this invoice.",
+        compute='_get_report_analytic_lines')
+
+    @api.multi
+    def lines_per_project(self):
+        """ Return analytic lines per project """
+
+        # inspired by odoo/addons/sale_layout/models/sale_layout.py
+        def grouplines(ordered_lines, sortkey):
+            """ Return lines grouped by category """
+            grouped_lines = []
+            for key, valuesiter in groupby(ordered_lines, sortkey):
+                group = {}
+                group['category'] = key
+                group['lines'] = list(v for v in valuesiter)
+                grouped_lines.append(group)
+            return grouped_lines
```

Comment:
I use this snippet myself on model level, that's a bit more general and more efficient because of the yielding. And you get recordsets at once, that's a bit more convenient to program with:

```python
    @api.multi
    def grouped(self, field='sale_order_id'):
        for key, group in itertools.groupby(
                self.sorted(lambda record: record[field]),
                lambda record: record[field]
        ):
            yield key, sum(group, self.browse([]))
```